src/meta/processors/admin/Balancer.cpp

```diff
@@ -752,25 +754,33 @@ int32_t Balancer::giveupLeaders(HostParts& leaderParts,
                                            }
                                            return leaderParts[l].size() < leaderParts[r].size();
                                        });
+
         // If peer can accept this partition leader, than host will transfer to the peer
-        auto& targetLeaders = leaderParts[*target];
-        int32_t targetLeaderSize = targetLeaders.size();
-        if (targetLeaderSize < hostBounds_[*target].second) {
-            it = sourceLeaders.erase(it);
-            targetLeaders.emplace_back(partId);
-            plan.emplace_back(spaceId, partId, source, *target);
-            LOG(INFO) << "giveup plan trans leader space: " << spaceId
-                      << " part: " << partId
-                      << " from " << source.host << ":" << source.port
-                      << " to " << target->host << ":" << target->port;
-            ++taskCount;
+        if (target != targets.end()) {
+            auto& targetLeaders = leaderParts[*target];
```

Comment:
I'm not sure if leaderParts[*target] may trigger re-hash, and invalid everything